File: java/src/main/java/nl/craftsmen/workshops/reactivemeetup/domain/geometry/Square.java
package nl.craftsmen.workshops.reactivemeetup.domain.geometry;

public class Square implements IShape {

    private String name;
    private final int width;

    public Square(String name, int width) {
        this.name = name;
        this.width = width;
    }

    @Override
    public double calculateArea() {
        return width * width;
    }

    @Override
    public double calculateCircumference() {
        return 4 * width;
    }

    @Override
    public String getName() {
        return name;
    }
}

File: javascript/src/domain/geometry/square.js
module.exports = class Circle {

    constructor(name, width) {
        this.name = name;
        this.width = width;
    }

    calculateArea() {
        return this.width * this.width;
    }

    calculateCircumference() {
        return 4 * this.width;
    }

    getName() {
        return this.name;
    }
}
